### PR TITLE
:card_file_box: [#4246] Expand AuthInfo model to capture mandate context

### LINKED ISSUE #4246: Capture authentication context data
Part of #3623

Currently the "machtigen" context data is stored as a simple JSON field, which doesn't capture everything that should be captured, and the shape is also not well documented. For this, some changes are needed:

- [x] Patch our OIDC library so that we can map properties with <kbd>.</kbd> characters too
- [x] Release new version of mozilla-django-oidc-db
- [x] Investigate removing the "django-solo" nature of the OIDC config model(s) (and do more with JSONField for config)
- [x] Establish structure of authentication context in models (`authentication.AuthInfo`)
    - Record/translate means (digid/eherkenning/...)
    - Record LOA
    - Record authorizee
    - Record representee
    - Record mandate context (service(s), role...)
- [x] Implement `AuthInfo.to_auth_context_data()` method and validate that against the JSON schema's (for robustness)
- [x] Extract necessary claims/context data in authentication plugins and save the data
- [x] Add context var to Objects API registration v1
- [x] Add registration var to Objects API registration v2
- [x] Update static variables to read from the appropriate place(s)
- [x] Update prefill to read from the appropriate place(s)

### PR DESCRIPTION
**Pull request by sergei-maertens**
Partly closes #4246

**Changes**

* Added fields for the (optional) legal subject (authorizee)
* Added fields for the (optional) acting subject (authorizee)
* Added check constraints to enforce data integrity
* Organized the admin in logical groups

The authentication context data model identifies two parties: the representee and the authorizee. There is always an authorizee - it *can* be the same actor as the representee if no mandate is in involved.

An authorizee has two aspects: legal subject and acting subject. There is always a legal subject. If no acting subject is provided, it is inferred from the legal subject.

So, a simple DigiD login in this model is represented by a legal subject authorizee of type BSN. We still keep storing this data in the attribute + value fields, but in the event of a mandate, we will store the additional information for the legal/acting subject.

**Checklist**

Check off the items that are completed or not relevant.

- Impact on features

  - [x] Checked copying a form
  - [x] Checked import/export of a form
  - [x] Config checks in the configuration overview admin page
  - [x] Problem detection in the admin email digest is handled

- Release management

  - [x] I have labelled the PR as "needs-backport" accordingly

- I have updated the translations assets (you do NOT need to provide translations)

  - [x] Ran `./bin/makemessages_js.sh.sh`
  - [x] Ran `./bin/compilemessages_js.sh`

- Commit hygiene

  - [x] Commit messages refer to the relevant Github issue
  - [x] Commit messages explain the "why" of change, not the how

### CODE REVIEW

**Comment by sergei-maertens** [comment]
For the reviewer, this relates to https://github.com/maykinmedia/authentication-context-schemas

File: src/openforms/authentication/models.py [hunk 1]
```diff
@@ -61,41 +68,154 @@ def hash_identifying_attributes(self, delay=False):
...
         on_delete=models.CASCADE,
         help_text=_("Submission related to this authentication information"),
         related_name="auth_info",
     )
-    machtigen = models.JSONField(
-        verbose_name=_("machtigen"),
+
+    # authentication details of the representee (without mandate/machtigen, this is the
+    # logged in actor themselves) are stored in the fields `plugin`, `attribute` and `value`.
+    # The match the (inferred) `representee.identifierType` and `representee.identifier`.
```
Comment (Viicos) [inline src/openforms/authentication/models.py:117]:
```suggestion
    # They match the (inferred) `representee.identifierType` and `representee.identifier`.
```

I guess?

File: src/openforms/authentication/models.py [hunk 2]
```diff
@@ -61,41 +68,154 @@ def hash_identifying_attributes(self, delay=False):
...
         ),
         blank=True,
         null=True,
     )
-    loa = models.TextField(
-        verbose_name=_("Level of assurance"),
+
+    # deprecated!
+    machtigen = models.JSONField(
+        verbose_name=_("machtigen"),
```
Comment (Viicos) [inline src/openforms/authentication/models.py:178]: I guess there's no way to extract things from this deprecated fields to set them to the mandate/authorizee fields?

**Review by Viicos — APPROVED** [review]

File: src/openforms/authentication/models.py (hunk 2 above)
Comment (sergei-maertens) [reply src/openforms/authentication/models.py:178]: even worse - the data stored in there is just plain wrong now :)

**Pull request merged**